fvm/fvm.go

```diff
@@ -33,7 +33,11 @@ func (vm *VirtualMachine) Run(ctx Context, proc Procedure, ledger state.Ledger)
 		state.WithMaxValueSizeAllowed(ctx.MaxStateValueSize),
 		state.WithMaxInteractionSizeAllowed(ctx.MaxStateInteractionSize))
 
-	return proc.Run(vm, ctx, st)
+	err := proc.Run(vm, ctx, st)
+	if err != nil {
+		return err
+	}
+	return st.Commit()
```

Comment:
Is this a bugfix?

Reply:
Not exactly, just moved the state commit to the function that created the state.